catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java

```diff
@@ -900,13 +889,13 @@ private String escapeSpecialCharacters(String searchPhrase) {
 
     private SolrQuery operationToQuery(String operation, String propertyName, String wkt) {
         String geoIndexName = getMappedPropertyName(propertyName, AttributeFormat.GEOMETRY, false);
-        return operationOnIndexToQuery(operation, geoIndexName, wkt);
-    }
-
-    private SolrQuery operationOnIndexToQuery(String operation, String indexName, String wkt) {
         if (StringUtils.isNotEmpty(wkt)) {
-            String geoQuery = indexName + ":\"" + operation + "(" + wkt + ")\"";
-            return getSolrQueryWithSort(geoQuery);
+            String geoQuery = geoIndexName + ":\"" + operation + "(" + wkt + ")\"";
+            Geometry pnt = getGeometry(wkt);
+            if (pnt != null) {
+                updateDistanceSort(propertyName, pnt.getCentroid());
+            }
+            return new SolrQuery(geoQuery);
         } else {
```

Comment:
You don't need this else. You could reduce the complexity by moving the If check and throw up to the top.